common/go-utils/src/vrs.rs

```diff
@@ -139,27 +98,23 @@ impl GoVersion {
             composed_version.push_str(pre.as_str());
         };
 
-        GoVersion::parse(&composed_version)
+        SemanticVersion::parse(&composed_version)
+            .map(GoVersion)
+            .map_err(GoVersionParseError::SemanticVersion)
     }
 }
 
 impl TryFrom<String> for GoVersion {
-    type Error = VersionParseError;
+    type Error = GoVersionParseError;
 
     fn try_from(val: String) -> Result<Self, Self::Error> {
         GoVersion::parse(&val)
     }
 }
 
-impl From<GoVersion> for String {
-    fn from(ver: GoVersion) -> Self {
-        format!("{ver}")
-    }
-}
-
 impl fmt::Display for GoVersion {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.0)
+        write!(f, "Go {}", self.0)
```

Comment:
Go versions and releases typically use the `go1.20.1` format upstream.

```suggestion
        write!(f, "go{}", self.0)
```

However, that's still not totally accurate. Prior to `go1.21.0`, the trailing `.0` wasn't included. If we want to use the historically accurate naming (I would prefer that), we'd need `1.20.0` to render as `go1.20` and `1.21.0` to render as `go1.21.0`.

Reply:
Agreed, this was a regression (which I attempted to justify here https://github.com/heroku/buildpacks-go/pull/245/commits/378b73ca045ed09b47dce45a9f7036e976216c02 ;-)) -- however, I definitely prefer displaying the actual version string over the parsed version.

The previous behavior was re-implemented [here](https://github.com/heroku/buildpacks-go/pull/245/commits/b0c757e75a972f9180c7d36de040783f63028261#diff-560d0a7bad62e4a300eba035429762f445bb85bb6ab9bc20f42b02c22545fb3cR96-R98), and the integration tests now verify this again https://github.com/heroku/buildpacks-go/pull/245/commits/a9bd12617724050031ec85cf9e70f5aa7c24c77a :)